pkg/auth/userregistry/identitymapper/lookup_test.go

```diff
@@ -154,3 +162,83 @@ func TestLookup(t *testing.T) {
 		}
 	}
 }
+
+type FakeMappingClient struct {
+	mappingRegistry mappingRegistry
+}
+
+func (f *FakeMappingClient) Create(obj *userapi.UserIdentityMapping) (*userapi.UserIdentityMapping, error) {
+	return f.mappingRegistry.CreateUserIdentityMapping(context.TODO(), obj)
+}
+func (f *FakeMappingClient) Update(obj *userapi.UserIdentityMapping) (*userapi.UserIdentityMapping, error) {
+	return f.mappingRegistry.UpdateUserIdentityMapping(context.TODO(), obj)
+}
+func (f *FakeMappingClient) Delete(name string, options *v1.DeleteOptions) error {
+	return f.mappingRegistry.DeleteUserIdentityMapping(context.TODO(), name)
+}
+func (f *FakeMappingClient) Get(name string, options v1.GetOptions) (*userapi.UserIdentityMapping, error) {
+	return f.mappingRegistry.GetUserIdentityMapping(context.TODO(), name, &options)
+}
+
+// Registry is an interface implemented by things that know how to store UserIdentityMapping objects.
+type mappingRegistry interface {
+	// GetUserIdentityMapping returns a UserIdentityMapping for the named identity
+	GetUserIdentityMapping(ctx apirequest.Context, name string, options *metav1.GetOptions) (*userapi.UserIdentityMapping, error)
+	// CreateUserIdentityMapping associates a user and an identity
+	CreateUserIdentityMapping(ctx apirequest.Context, mapping *userapi.UserIdentityMapping) (*userapi.UserIdentityMapping, error)
+	// UpdateUserIdentityMapping updates an associated user and identity
+	UpdateUserIdentityMapping(ctx apirequest.Context, mapping *userapi.UserIdentityMapping) (*userapi.UserIdentityMapping, error)
+	// DeleteUserIdentityMapping removes the user association for the named identity
+	DeleteUserIdentityMapping(ctx apirequest.Context, name string) error
+}
+
+// Storage is an interface for a standard REST Storage backend
+// TODO: move me somewhere common
+type mappingStorage interface {
+	rest.Getter
+	rest.Deleter
+
+	Create(ctx apirequest.Context, obj runtime.Object, _ bool) (runtime.Object, error)
+	Update(ctx apirequest.Context, name string, objInfo rest.UpdatedObjectInfo) (runtime.Object, bool, error)
+}
+
+// storage puts strong typing around storage calls
+type storage struct {
+	mappingStorage
+}
+
+// NewRegistry returns a new Registry interface for the given Storage. Any mismatched
```

Comment:
> Is all of this registry code in this test just to make the old interface based tests work?

Yeah.  I'll let the testing debt be cleaned up separately.  The interfaces in the real code are now correct and the existing testing registries could be removed or they could be left.  I have no preference.